zc_common/remote_resource/request.py

```diff
@@ -80,19 +80,30 @@ def add_items_from_data(self):
         map(lambda x: self.append(RemoteResourceWrapper(x)), self.data)
 
 
-def get_route_from_fk(resource_type, pk=None):
-    """Gets a fully qualified URL for a given resource_type, pk"""
-    routes = requests.get(zc_settings.GATEWAY_ROOT_PATH).json()
-
-    for route in routes.iterkeys():
-        if 'resource_type' in routes[route] and routes[route]['resource_type'] == resource_type:
-            if isinstance(pk, (list, set)):
-                expanded = '{}?filter[id__in]={}'.format(expand(route, {}), ','.join([str(x) for x in pk]))
-            else:
-                expanded = expand(route, {'id': pk})
-            return '{0}{1}'.format(routes[route]['domain'], expanded)
-
-    raise RouteNotFoundException('No route for resource_type: "{0}"'.format(resource_type))
+class RouteManager(object):
```

Comment:
@birkholz I understand the performance improvement you want to make by this `RouteManager`. However, the implementation is a bit inflexible. How am supposed to conveniently populate the cache?

You implementation assumes the `GATEWAY_ROOT_PATH` is always accessible which is false when running unit tests. However, by caching the first request, at test which hits `GATEWAY_ROOT_PATH` will then get its response cached and thus have unit tests running without blocking. I can't test my changes now because of this problem.